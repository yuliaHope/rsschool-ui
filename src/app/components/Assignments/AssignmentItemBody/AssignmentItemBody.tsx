import * as React from 'react';
import { CardBody } from 'reactstrap';

type AssignmentItemBodyProps = {
    title: string;
    urlToDescription?: string;
};

const AssignmentItemBody = (props: AssignmentItemBodyProps) => {
    const { title, urlToDescription } = props;
    return (
        <CardBody>
            <h5 className="card-title">{title}</h5>
            <p className="card-text">{urlToDescription}</p>
        </CardBody>
    );
};

export default AssignmentItemBody;
