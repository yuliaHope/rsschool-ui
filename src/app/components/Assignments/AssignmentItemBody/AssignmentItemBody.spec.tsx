import * as React from 'react';
import { shallow } from 'enzyme';
import AssignmentItemBody from './AssignmentItemBody';

describe('Assignment Item Body', () => {
    const props = {
        title: '',
        urlToDescription: '',
    };

    it('render with no data', () => {
        const output = shallow(<AssignmentItemBody {...props} />);
        expect(output).toMatchSnapshot();
    });

    it('render with data', () => {
        const nextProps = {
            ...props,
            title: 'title',
            urlToDescription: 'urlToDescription',
        };
        const output = shallow(<AssignmentItemBody {...nextProps} />);
        expect(output).toMatchSnapshot();
    });
});
