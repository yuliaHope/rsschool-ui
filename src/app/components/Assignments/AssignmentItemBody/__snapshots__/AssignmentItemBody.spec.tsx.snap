// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assignment Item Body render with data 1`] = `
<CardBody
  tag="div"
>
  <h5
    className="card-title"
  >
    title
  </h5>
  <p
    className="card-text"
  >
    urlToDescription
  </p>
</CardBody>
`;

exports[`Assignment Item Body render with no data 1`] = `
<CardBody
  tag="div"
>
  <h5
    className="card-title"
  />
  <p
    className="card-text"
  />
</CardBody>
`;
